src/main/kotlin/app/Configurator.kt

```diff
@@ -0,0 +1,178 @@
+// Copyright 2017 Sourcerer Inc. All Rights Reserved.
+// Author: Anatoly Kislov (anatoly@sourcerer.io)
+
+package app
+
+import app.utils.Options
+import app.utils.PasswordValidator
+import app.utils.UsernameValidator
+import java.io.File
+import java.io.IOException
+import org.apache.commons.configuration.ConfigurationException
+import org.apache.commons.configuration.PropertiesConfiguration
+
+/**
+ * Configurator is a singleton class that manages configuration files and
+ * values of non-command specific options.
+ */
+object Configurator {
+    val configFileName = "sourcerer.properties"
+    val configUsername = "username"
+    val configPassword = "password"
+    val configSilent = "silent"
+
+    // Options levels are presented in priority decreasing order.
+    private var current: Options = Options()  // Command-line arguments.
+    private var config: Options = Options()  // Global user defined config file.
+    private val default: Options  // Default values.
+        get() {
+            val default = Options()
+            default.silent = false
+            return default
+        }
+
+    val options: Options  // Final options that will be used by app.
+        get() = mergeLevels()
+
+    // User directory path.
+    val userDir = try {
+        System.getProperty("user.home")
+    }
+    catch (e: SecurityException) { null }
+
+    init {
+        loadUserLevelConfig()
+    }
+
+    // Merges different levels of options into one options object in the next
+    // order: current, local, user, default.
+    private fun mergeLevels(): Options {
+        val levels = arrayListOf(current, config, default)
+        val merged = Options()
+
+        for (level in levels) {
```

Comment:
It could be done with reflection. But it will increase the overall size of the app and can have some performance drawback.